src/components/axes/abstractAxis.ts

```diff
@@ -416,18 +426,19 @@ export module Axis {
         return this._orientation;
       } else {
         var newOrientationLC = newOrientation.toLowerCase();
-        if (newOrientationLC !== "top" &&
-            newOrientationLC !== "bottom" &&
-            newOrientationLC !== "left" &&
-            newOrientationLC !== "right") {
-          throw new Error("unsupported orientation");
-        }
+        AbstractAxis.verifyAxisOrientation(newOrientationLC);
         this._orientation = newOrientationLC;
         this._invalidateLayout();
         return this;
       }
     }
 
+    private static verifyAxisOrientation(orientation: string) {
```

Comment:
Maybe the lower-case checking can happen here? And then this method can return a lowercased version of the orientation.